src/main/java/org/openrewrite/maven/AbstractRewriteMojo.java

```diff
@@ -67,6 +69,9 @@ public abstract class AbstractRewriteMojo extends ConfigurableRewriteMojo {
     @Parameter(property = "rewrite.resolvePropertiesInYaml", defaultValue = "true")
     protected boolean resolvePropertiesInYaml;
 
+    @Parameter(property = "rewrite.exportDatatables", defaultValue = "false")
+    protected boolean exportDatatables;
```

Comment:
I understand why you added the property here, as this class contains the `runRecipe` method. But looking at the wider class hierarchy I worry that this option then also shows up in the documentation for the CycloneDxBomMojo and RewriteDiscoverMojo: https://openrewrite.github.io/rewrite-maven-plugin/plugin-info.html

![image](https://github.com/openrewrite/rewrite-maven-plugin/assets/1027334/c7c4d7de-82be-4580-accc-fd72bc5e90cd)

Perhaps we can revisit the hierarchy to only show this option where applicable? Perhaps push it down into `AbstractRewriteRunMojo` ?

Reply:
Ideally this datatable export feature would be available for both regular runs and dry runs, which was my main reasoning for positioning it in `AbstractRewriteMojo` class as it happens to be the first (and only) parent class shared between those. 

I understand your concern regarding the other two classes though. We could consider moving the parameter and corresponding logic down into the `AbstractRewriteRunMojo` and `AbstractRewriteDryRunMojo` individually, but that would lead to some unfortunate duplication.

Reply:
Yeah there's not really an immediately clear satisfactory way to handle this; duplication is unfortunate as well, but so is unused documented options. :/

Reply:
So, looking into this a little further, both the `AbstractRewriteRunMojo` and `AbstractRewriteDryRunMojo` would have to override the `runRecipe` method from `AbstractRewriteMojo`, which would result in the `AbstractRewriteMojo` `runRecipe` implementation being dead code, as no other subclasses use that method. Making that method abstract is also not ideal, as it would force the `CycloneDxBomMojo` and `RewriteDiscoverMojo` classes to implement it unnecessarily. 

The best solution I can think of is to create another new Mojo class that extends `AbstractRewriteMojo` that `AbstractRewriteRunMojo` and `AbstractRewriteDryRunMojo` both extend. We could move any functionality that's only used in the two run mojo's there (including the `runRecipe` method and exportDatatables parameter), while keeping the other mojo's the same. This would be a bigger change but would probably be an overall better architecture. 

Definitely interested to hear your thoughts on this, or if you have any other ideas/approaches.

Reply:
That might indeed be the best way to go about it; but we have painted ourselves into the corner a bit on what to name that new class. 😅 How about `BaseRewriteRunMojo` ? Open to alternatives!

Reply:
Alright, did that class refactor. Opted to go with `AbstractRewriteBaseRunMojo` as the name of the new class, but also open to alternative naming.